server/db-middleware/general-api.js

```diff
@@ -61,5 +61,22 @@ apiModule.getStudentsByGroup = function (groupId, studProj) {
     .lean();
 };
 
+apiModule.getFilteredStudentsByTeacher = function (teacherId) {
```

Comment:
The main difference between this method's result and getStudentsByTeacher's result is it's structure.
Here we have flat result, so maybe better name getStudentsByTeacherFlat ?